src/UI/Implementation/Component/Chart/Speedo/Speedo.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+/* Copyright (c) 2017 Ralph Dittrich <dittrich@qualitus.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Implementation\Component\Chart\Speedo;
+
+use ILIAS\UI\Component as C;
+use ILIAS\UI\Implementation\Component\ComponentHelper;
+/**
+ * Class Speedo
+ * @package ILIAS\UI\Implementation\Component\Chart\Speedo
+ */
+class Speedo implements C\Chart\Speedo\Speedo {
+    use ComponentHelper;
+
+    /**
+     * @var int
+     */
+    protected $maximum;
+
+    /**
+     * @var int
+     */
+    private $minimum;
+
+    /**
+     * @var int
+     */
+    protected $score;
+
+    /**
+     * @var int
+     */
+    protected $diagnostic;
+
+    /**
+     * @inheritdoc
+     */
+    public function __construct($maximum, $score, $minimum = null, $diagnostic = null)
+    {
+        $this->checkIntArg("maximum", $maximum);
+        $this->maximum = $maximum;
+        $this->checkIntArg("score", $score);
+        $this->score = $this->getSafe($score);
+
+        if($minimum != null) {
+            $this->checkIntArg("minimum", $minimum);
+            $this->minimum = $this->getSafe($minimum);
+        } else {
+            $this->checkIntArg("minimum", $maximum);
+            $this->minimum = $this->getSafe($maximum);
+        }
+        if($diagnostic != null) {
+            $this->checkIntArg("diagnostic", $diagnostic);
+            $this->diagnostic = $this->getSafe($diagnostic);
+        }
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getMaximum()
+    {
+        return (isset($this->maximum) ? $this->maximum : 100);

```

Comment:
`isset($this->maximum)` will always be true under sane circumstances (no reflection classes, odd inheritances of this class, ...), right?